src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -1755,6 +1789,26 @@ private void emitNoResolvedType(NoType type, String defaultEmit) {
       }
     }
 
+    private String getDisplayNameFromType(JSType type) {
+      String displayName = type.getDisplayName();
+      // In partial mode, closure doesn't know the correct name of imported symbols, if the name
+      // matches one in the precomputed map, replace it with the original declared name
+      // The displayName can be of the form foo.bar, but the symbol that was goog required was just
+      // foo, so just replace the part of the display name before the first period
+      List<String> displayNameParts = new ArrayList<>(Arrays.asList(displayName.split("\\.")));
+      for (int i = 1; i <= displayNameParts.size(); i++) {
+        List<String> displayNamePrefix = displayNameParts.subList(0, i);
+        String baseDisplayName = Joiner.on(".").join(displayNamePrefix);
+        if (importRenameMap.containsKey(baseDisplayName)) {
+          displayNamePrefix.clear();
+          displayNamePrefix.add(importRenameMap.get(baseDisplayName));
```

Comment:
This seems to do some work on displayNamePrefix but then never use it (?)

Reply:
`displayNamePrefix` is a sublist of `displayNameParts`, so modifiying it modifies `displayNameParts`.  Added a comment.